pkg/java/jar.go

```diff
@@ -96,25 +112,34 @@ func md5File(file fs.File) (string, int64, error) {
 }
 
 func md5Bytecode(r *zip.ReadCloser, classLocation string) (string, error) {
-	c, err := r.Open(classLocation + ".class")
+	buf, err := readClassBytes(r, classLocation)
 	if err != nil {
 		return "", err
 	}
 
-	buf := new(bytes.Buffer)
-	_, err = buf.ReadFrom(c)
+	h, err := HashClassInstructions(buf.Bytes())
 	if err != nil {
 		return "", err
 	}
 
-	h, err := HashClassInstructions(buf.Bytes())
+	return h, nil
+}
+
+func readClassBytes(r *zip.ReadCloser, classLocation string) (*bytes.Buffer, error) {
+	c, err := r.Open(classLocation + ".class")
 	if err != nil {
-		return "", err
+		return nil, err
+	}
+
+	buf := new(bytes.Buffer)
+	_, err = buf.ReadFrom(c)
+	if err != nil {
```

Comment:
nit: let's inline this
```suggestion
	if _, err := buf.ReadFrom(c); err != nil {
```